bundles/CoreBundle/src/Migrations/Version20230124120641.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\CoreBundle\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+final class Version20230124120641 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Covert column types to JSON for columns containing only JSON data';
+    }
+
+    public function up(Schema $schema): void
+    {
+        if ($schema->getTable('classificationstore_keys')->hasColumn('definition') === true) {
+            $this->addSql("ALTER TABLE `classificationstore_keys` MODIFY COLUMN `definition` json;");
```

Comment:
What happens if this is not valid JSON currently? How does failure look like here?

Reply:
In the columns which we are changed should be only valid JSON data stored.

Reply:
> should be

Of course, my point is what if it isn't, do we need to handle that? If data is invalid JSON and the migration fails here, it's pretty bad for the user, maybe we can do something to prevent that.

For example, we could try running a query which does the validation on all tables, like so

```sql
SELECT id, JSON_VALID(definition) FROM classificationstore_keys;
```

for all tables and, only once we confirm they **all** produce no invalid JSON do we allow the migration to run.

Reply:
Just tried having invalid JSON in the column and it produces

```
Invalid JSON text: "Invalid value." at position 1 in value for column '#sql-1_10e.definition'.
```

Reply:
In general I see your point that this could cause issues. 
However we changed the tables here where the application is expecting data in JSON format (e.g. https://github.com/pimcore/pimcore/blob/11.x/models/DataObject/Classificationstore/Service.php#L63)
Therefore if there will be invalid JSON data stored in these columns then the application would not work anymore